File: 2-BackEnd/0-PHP/B - PHP avançado/02-RESTful-API-PHP-PDO/src/Product.php
<?php
namespace App;

class Product
{
    public function __construct(
        public readonly int    $id,
        public readonly string $name,
        public readonly string $description,
        public readonly float  $price,
        public readonly int    $stock,
        public readonly string $created_at = '',
        public readonly string $updated_at = '',
    ) {}

    public function toArray(): array
    {
        return [
            'id'          => $this->id,
            'name'        => $this->name,
            'description' => $this->description,
            'price'       => $this->price,
            'stock'       => $this->stock,
            'created_at'  => $this->created_at,
            'updated_at'  => $this->updated_at,
        ];
    }

    public static function fromArray(array $row): self
    {
        return new self(
            id:          (int)$row['id'],
            name:        $row['name'],
            description: $row['description'] ?? '',
            price:       (float)$row['price'],
            stock:       (int)$row['stock'],
            created_at:  $row['created_at'] ?? '',
            updated_at:  $row['updated_at'] ?? '',
        );
    }
}

File: 2-BackEnd/12-Python-FastAPI/B - Python-FastAPI avançado/01-FastAPI-SQLAlchemy-Postgres/app/models/product.py
from decimal import Decimal
from typing import Optional

from sqlalchemy import ForeignKey, Numeric, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base


class Product(Base):
    __tablename__ = "products"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String(120), nullable=False)
    sku: Mapped[str] = mapped_column(String(40), unique=True, nullable=False, index=True)
    description: Mapped[Optional[str]] = mapped_column(Text(), nullable=True)
    price: Mapped[Decimal] = mapped_column(Numeric(10, 2), nullable=False)
    stock: Mapped[int] = mapped_column(default=0, nullable=False)
    category_id: Mapped[int] = mapped_column(ForeignKey("categories.id"), nullable=False)
    category: Mapped["Category"] = relationship(back_populates="products")
